### PR TITLE
Implementation of Euclideans Algorithm in Kotlin.

### PR DESCRIPTION
**Pull request by olaven**
This is my first time contributing to this project. 
Hopefully, I've done everything correctly. 
Otherwise, let me know, and I will fix it as fast as possible 🚂🚋🚋  

Thank you for creating such an inspiring project🌦

### CODE REVIEW

File: contents/euclidean_algorithm/code/kotlin/EuclideanSub.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import kotlin.math.absoluteValue
+
+fun euclidSub(a: Int, b: Int): Int {
+    var a = a.absoluteValue
+    var b = b.absoluteValue
+
+    while(a != b) {
```
Comment (berquist) [inline contents/euclidean_algorithm/code/kotlin/EuclideanSub.kt]:
```suggestion
    while (a != b) {
```

File: contents/euclidean_algorithm/code/kotlin/EuclideanMod.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    while (b != 0) {
+        val tmp = b
+        b = a % b
+        a = tmp
+    }
+
+    return a
+}
+
+fun main() {
```
Comment (berquist) [inline contents/euclidean_algorithm/code/kotlin/EuclideanMod.kt]: In Kotlin, `main` should always take an array of arguments:
```suggestion
fun main(args: Array<String>) {
```

File: contents/euclidean_algorithm/code/kotlin/EuclideanSub.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        if (a > b) a -= b
+        else b -= a
+    }
+
+    return a
+}
+
+fun main() {
```
Comment (berquist) [inline contents/euclidean_algorithm/code/kotlin/EuclideanSub.kt]: Same as above.

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 4]
```diff
@@ -15,6 +15,8 @@ The algorithm is a simple way to find the *greatest common divisor* (GCD) of two
 [import:18-31, lang="c_cpp"](code/c++/euclidean.cpp)
 {% sample lang="java" %}
 [import:3-16, lang="java"](code/java/EuclideanAlgo.java)
+{% sample lang="kotlin" %}
+[import:1-13, lang="kotlin"](code/kotlin/EuclideanSub.kt)
```
Comment (berquist) [inline contents/euclidean_algorithm/euclidean_algorithm.md]: I would have this and the other import not include the `import kotlin.math.absoluteValue` line for the book display.

**Comment by berquist** [comment]
Oh, there is one other thing I noticed that isn't something you need to change for this PR. I see that you're making the PR from the master branch of your fork. Once you start making multiple PRs for a project (not just this one, any one), or if you want to update your fork's master from the upstream master, your life will become hell. You should make a branch for your code and do the PR from there: https://github.com/algorithm-archivists/algorithm-archive/wiki/How-to-Contribute#step-32---branching

**Comment by olaven** [comment]
Thank you for the improvements and advice😄 I greatly appreaciate it. 
I believe everything is resolved with #589. If not, let me know.

File: contents/euclidean_algorithm/code/kotlin/Euclidean.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    while (b != 0) {
+        val tmp = b
+        b = a % b
+        a = tmp
+    }
+
+    return a
+}
+
+fun main() {
```
Comment (berquist) [inline contents/euclidean_algorithm/code/kotlin/Euclidean.kt]: Forgot something:
```suggestion
fun main(args: Array<String>) {
```

**Comment by berquist** [comment]
Looks good, thanks again! I'm even gonna leave all the emoji in the commit log.

**Pull request merged**